utildb/gen_deleted_accounts.go

```
@@ -59,100 +67,284 @@ func readAccounts(ch chan proxy.ContractLiveliness, deleteHistory *map[common.Ad
 
 // genDeletedAccountsTask process a transaction substate then records self-destructed accounts
 // and resurrected accounts to a database.
-func genDeletedAccountsTask(
-	tx *substate.Transaction,
-	processor *executor.TxProcessor,
-	ddb *substate.DestroyedAccountDB,
-	deleteHistory *map[common.Address]bool,
-	cfg *utils.Config,
-) error {
+func genDeletedAccountsTask(tx *substate.Transaction, processor executor.TxProcessor, cfg *utils.Config, log logger.Logger) ([]proxy.ContractLiveliness, error) {
 	ch := make(chan proxy.ContractLiveliness, channelSize)
 	var statedb state.StateDB
 	var err error
 	ss := substatecontext.NewTxContext(tx.Substate)
 
 	statedb, err = state.MakeOffTheChainStateDB(ss.GetInputState(), tx.Block, state.NewChainConduit(cfg.ChainID == utils.EthereumChainID, utils.GetChainConfig(cfg.ChainID)))
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	//wrapper
-	statedb = proxy.NewDeletionProxy(statedb, ch, cfg.LogLevel)
+	statedb = proxy.NewDeletionProxy(statedb, ch, log)
 
 	_, err = processor.ProcessTransaction(statedb, int(tx.Block), tx.Transaction, ss)
 	if err != nil {
-		return nil
+		return nil, nil
```

Should we return the error here?